code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -482,53 +482,54 @@
+ */
+/mob/living/simple_animal/bot/proc/scan(list/scan_types, old_target, scan_range = DEFAULT_SCAN_RANGE)
+	var/turf/current_turf = get_turf(src)
+	if(!current_turf)
 		return
-	var/list/adjacent = T.get_atmos_adjacent_turfs(1)
+	var/list/adjacent = current_turf.get_atmos_adjacent_turfs(1)
 	if(shuffle) //If we were on the same tile as another bot, let's randomize our choices so we dont both go the same way
 		adjacent = shuffle(adjacent)
 		shuffle = FALSE
-	for(var/scan in adjacent)//Let's see if there's something right next to us first!
-		if(check_bot(scan)) //Is there another bot there? Then let's just skip it
-			continue
-		if(isturf(scan_type)) //If we're lookeing for a turf we can just run the checks directly!
-			var/final_result = checkscan(scan,scan_type,old_target)
+
+	for(var/scan_type_list in scan_types)
+
+		for(var/turf/scan as anything in adjacent) //Let's see if there's something right next to us first!
+			if(check_bot(scan)) //Is there another bot there? Then let's just skip it
+				continue
+			var/final_result = checkscan(scan, scan_types, old_target)
+			if(final_result)
+				return final_result
+
+		for(var/turf/scanned_turfs as anything in shuffle(view(scan_range, src))-adjacent) //Search for something in range, minus what we already checked.
+			if(check_bot(scanned_turfs)) //Is there another bot there? Then let's just skip it
+				continue
+			var/final_result = checkscan(scanned_turfs, scan_types, old_target)
 			if(final_result)
 				return final_result
+
+/mob/living/simple_animal/bot/proc/checkscan(atom/scan, list/scan_types, old_target)
+	for(var/scan_type in scan_types)
+		if(!istype(scan, scan_type)) //Check that the thing we found is the type we want!
+			continue //If not, keep searching!
+		if((REF(scan) in ignore_list) || (scan == old_target)) //Filter for blacklisted elements, usually unreachable or previously processed oness
+			continue
+
+		var/scan_result = process_scan(scan) //Some bots may require additional processing when a result is selected.
+		if(scan_result)
+			return scan_result
 		else
-			var/turf/turfy = scan
-			for(var/deepscan in turfy.contents)//Check the contents since adjacent is turfs
-				var/final_result = checkscan(deepscan,scan_type,old_target)
-				if(final_result)
-					return final_result
-	for (var/scan in shuffle(view(scan_range, src))-adjacent) //Search for something in range!
-		var/final_result = checkscan(scan,scan_type,old_target)
-		if(final_result)
-			return final_result
-
-/mob/living/simple_animal/bot/proc/checkscan(scan, scan_type, old_target)
-	if(!istype(scan, scan_type)) //Check that the thing we found is the type we want!
-		return FALSE //If not, keep searching!
-	if((REF(scan) in ignore_list) || (scan == old_target)) //Filter for blacklisted elements, usually unreachable or previously processed oness
-		return FALSE
+			continue //The current element failed assessment, move on to the next.
```

Comment:
"else continue" at the end of the loop?

if it's intentional to be explicit feel free to ignore